### DISCUSSION TITLE
How could I get log dict?

### DISCUSSION THREAD

**Discussion by yllgl** [Lightning-AI/pytorch-lightning#7670] [General]
I logged the train_loss variable with self.log, and I want to access it. I think it should be recorded in self._results. But when I access self._results, nothing is recorded.
```python
def training_step(self,batch, batch_idx):
        x,y=batch
        x= x.float()
        y=y.float()
        x=self(x)
        loss_fn=nn.MSELoss()
        loss=loss_fn(x,y)
        self.log('train_loss', loss,on_epoch=True)
        return loss
def on_epoch_end(self) -> None:
        print(self._results['train_loss'])
```
How could I get logging information?

**Comment by ethanwharris** [comment]
Hi @yllgl, you should be able to access the logged metrics through `self.trainer.callback_metrics`. Hope that helps :smiley:
